src/register/EmbedableRegistrationForm.jsx

```diff
@@ -0,0 +1,583 @@
+import React, {
+  useEffect, useMemo, useState,
+} from 'react';
+import { connect } from 'react-redux';
+
+import { getConfig, snakeCaseObject } from '@edx/frontend-platform';
+import { sendPageEvent } from '@edx/frontend-platform/analytics';
+import {
+  getCountryList, getLocale, useIntl,
+} from '@edx/frontend-platform/i18n';
+import { Form, StatefulButton } from '@edx/paragon';
+import PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet';
+
+import ConfigurableRegistrationForm from './ConfigurableRegistrationForm';
+import {
+  clearRegistertionBackendError,
+  clearUsernameSuggestions,
+  fetchRealtimeValidations,
+  registerNewUser,
+} from './data/actions';
+import {
+  COUNTRY_CODE_KEY,
+  COUNTRY_DISPLAY_KEY,
+  FORM_SUBMISSION_ERROR,
+} from './data/constants';
+import { registrationErrorSelector, validationsSelector } from './data/selectors';
+import {
+  emailRegex, getSuggestionForInvalidEmail, urlRegex, validateCountryField, validateEmailAddress,
+} from './data/utils';
+import messages from './messages';
+import RegistrationFailure from './RegistrationFailure';
+import { EmailField, UsernameField } from './registrationFields';
+import {
+  FormGroup, PasswordField,
+} from '../common-components';
+import { getThirdPartyAuthContext } from '../common-components/data/actions';
+import {
+  fieldDescriptionSelector,
+} from '../common-components/data/selectors';
+import {
+  DEFAULT_STATE,
+  LETTER_REGEX, NUMBER_REGEX, REDIRECT,
+} from '../data/constants';
+import {
+  getAllPossibleQueryParams, isHostAvailableInQueryParams, setCookie,
+} from '../data/utils';
+
+const EmbedableRegistrationForm = (props) => {
+  const {
+    backedUpFormData,
+    backendCountryCode,
+    backendValidations,
+    fieldDescriptions,
+    registrationError,
+    registrationErrorCode,
+    registrationResult,
+    submitState,
+    usernameSuggestions,
+    validationApiRateLimited,
+    // Actions
+    getRegistrationDataFromBackend,
+    validateFromBackend,
+    clearBackendError,
+  } = props;
+
+  const { formatMessage } = useIntl();
+  const countryList = useMemo(() => getCountryList(getLocale()), []);
+  const queryParams = useMemo(() => getAllPossibleQueryParams(), []);
+  const registrationEmbedded = isHostAvailableInQueryParams();
```

Comment:
this form is accessible for embedded experience only. I don't think we need this. 
we can pass true hardcoded in the `ConfigurableRegistrationFields` component.